HW1/P4/P4.py

```diff
@@ -0,0 +1,64 @@
+import findspark
+findspark.init()
+import pyspark
+import json
+from P4_bfs import bfs
+import time
+sc = pyspark.SparkContext(appName="P4")
+start = time.time()
+c = sc.textFile('source.csv')
+
+# This function creates a tuple
+# input----word: the word in the .txt file
+# ouput----a tuple in the required format
+def preprocess_mapper(x):
+    x = x.split('","')   
+    if len(x) == 2:
+        character = x[0][1:]
+        comic = x[1][:-1]
+        return (comic, character)
+    return (-1, -1)
+
+movie_characters_RDD = c.map(preprocess_mapper).filter(lambda (k, v): v != -1).groupByKey().mapValues(lambda x: list(x))
```

Comment:
The code might be easier to read if you split this over multiple lines. But I think I get it.